diff --git a/drivers/painter/comms/qp_comms_spi.h b/drivers/painter/comms/qp_comms_spi.h
index c39ea95f72..156cb6fd70 100644
--- a/drivers/painter/comms/qp_comms_spi.h
+++ b/drivers/painter/comms/qp_comms_spi.h
@@ -1,5 +1,7 @@
 // Copyright 2021 Nick Brassel (@tzarc)
+// Copyright Pablo Martinez (@elpekenin) <elpekenin@elpekenin.dev>
 // SPDX-License-Identifier: GPL-2.0-or-later
+
 #pragma once

 #ifdef QUANTUM_PAINTER_SPI_ENABLE
@@ -45,6 +51,8 @@ void     qp_comms_spi_dc_reset_bulk_command_sequence(painter_device_t device, co

 extern const painter_comms_with_command_vtable_t spi_comms_with_dc_vtable;

+extern const painter_comms_with_command_vtable_t spi_comms_with_dc_single_byte_vtable;
+
 #    endif // QUANTUM_PAINTER_SPI_DC_RESET_ENABLE

 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
